apps/emqx/src/bhvrs/emqx_external_log_handler.erl

```diff
@@ -0,0 +1,28 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+-module(emqx_external_log_handler).
+
+-callback tr_handlers(Conf :: map()) ->
+    [{handler, logger:handler_id(), module(), logger:handler_config()}].
+
+-export([tr_handlers/1]).
+
+tr_handlers(Conf) ->
+    lists:flatmap(fun(M) -> M:tr_handlers(Conf) end, tr_modules()).
+
+tr_modules() ->
+    [emqx_otel_log_handler].
```

Comment:
This probably should be better decoupled from the main `emqx` app, but I tried to keep it simple for now.

Reply:
can we install the handler at runtime? i.e. in emqx_opentelemetry application

Reply:
@zmstone, yes, but it would mean that before` emqx_opentelemetry` app is started, logs won't be sent to the external otel collector/backend. Unless we start emqx_opentelemetry earlier than any other app.

Reply:
BTW, current `otel_log_handler` implementation anyway doesn't allow to start a handler at early stage (when kernel logger starts). I think we have two options for now:
- change otel_log_handler to start under kernel's `logger_sup`, start accepting log event writes to the table but postpone 'full init' before kernel is fully started. 
   Advantages: the handler can work similarly to native log handlers with no log event losses.
   Drawbacks:  - have to rely on some kernel internals (logger_sup),
                        - more complicated log_handler design is required
 - keep otel_log_handler as is and accept the fact that it can't be initialized at early boot stage, meaning some logs will always be missing (probably up to the point when emqx_conf app initiates).

Reply:
Note: my previous comment is not absolutely correct. There was actually a deadlock trying to start ssl app (opentelemetry_exporter dependency) in adding_handler/1 logger cb. I'm fixing that now in opentelemetry lib, so starting a handler under kernel logger_sup shouldn't be required.

Reply:
I don't think it's so critical to have the boot logs sent to otel ?

Reply:
Moved everything to `emqx_opentelemetry` app